pkg/common/cns-lib/volume/listview.go

```diff
@@ -126,11 +131,30 @@ func (l *ListViewImpl) AddTask(ctx context.Context, taskMoRef types.ManagedObjec
 	})
 	log.Debugf("task %+v added to map", taskMoRef)
 
-	err := l.listView.Add(l.ctx, []types.ManagedObjectReference{taskMoRef})
+	req := types.ModifyListView{
+		This: l.listView.Reference(),
+		Add:  []types.ManagedObjectReference{taskMoRef},
+	}
+	response, err := methods.ModifyListView(ctx, l.listView.Client(), &req)
 	if err != nil {
 		l.taskMap.Delete(taskMoRef)
-		return logger.LogNewErrorf(log, "failed to add task to ListView. error: %+v", err)
+		return fmt.Errorf("%w. err: %v", ErrListViewAddition, err)
```

Comment:
Is ModifyListView only doing a "Add Task" ? 
If it is performing a Add & a Remove operation, we should probably rename `ErrListViewAddition` --> `ErrListViewModification`

Reply:
So, both `Add(refs [])` and `Remove(refs [])` methods from the ListView API internally call the same `methods.ModifyListView()` by setting the appropriate field in the request. 

I replaced the Add() but haven't replaced Remove() with methods.ModifyListView(remove []). 
In the govmomi version we are consuming currently, ModifyListView() doesn't return expired tasks in case of removal. See,

1. https://github.com/vmware/govmomi/blob/release-0.32/simulator/view_manager.go#L271-L273
2. https://github.com/vmware/govmomi/blob/main/simulator/registry.go#L473-L480

So, we can keep it to `ErrListViewAddition`.